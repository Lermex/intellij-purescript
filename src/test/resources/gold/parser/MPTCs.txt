Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('Main')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSImportDeclarationImpl(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(ImportModuleName)
          PSASTWrapperElement(Qualified)
            PsiElement(proper name)('Prelude')
          PsiElement(whitespace)('\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSImportDeclarationImpl(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(ImportModuleName)
          PSASTWrapperElement(Qualified)
            PsiElement(proper name)('Control')
            PsiElement(.)('.')
            PsiElement(proper name)('Monad')
            PsiElement(.)('.')
            PsiElement(proper name)('Eff')
            PsiElement(.)('.')
            PsiElement(proper name)('Console')
          PsiElement(whitespace)(' ')
        PsiElement(()('(')
        PSASTWrapperElement(PositionedDeclarationRef)
          PSASTWrapperElement(ValueRef)
            PsiElement(identifier)('log')
        PsiElement())(')')
        PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(TypeClassDeclaration)
        PsiElement(class)('class')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(Qualified)
            PSProperNameImpl(ProperName)
              PsiElement(proper name)('NullaryTypeClass')
              PsiElement(whitespace)(' ')
        PSASTWrapperElement(Implies)
          <empty list>
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(TypeDeclaration)
          PSASTWrapperElement(TypeAnnotationName)
            PsiElement(identifier)('greeting')
            PsiElement(whitespace)(' ')
          PsiElement(::)('::')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeConstructor)
                PSASTWrapperElement(Qualified)
                  PSProperNameImpl(ProperName)
                    PsiElement(proper name)('String')
                    PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(TypeInstanceDeclaration)
        PsiElement(instance)('instance')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(GenericIdentifier)
          PsiElement(identifier)('nullaryTypeClass')
          PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(Qualified)
            PSProperNameImpl(ProperName)
              PsiElement(proper name)('NullaryTypeClass')
              PsiElement(whitespace)(' ')
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(ValueDeclaration)
          PsiElement(identifier)('greeting')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(NamedBinder)
            <empty list>
          PsiElement(=)('=')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(StringLiteral)
                PsiElement(string)('"Hello, World!"')
                PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(TypeClassDeclaration)
        PsiElement(class)('class')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(Qualified)
            PSProperNameImpl(ProperName)
              PsiElement(proper name)('Coerce')
              PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('b')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(Implies)
          <empty list>
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(TypeDeclaration)
          PSASTWrapperElement(TypeAnnotationName)
            PsiElement(identifier)('coerce')
            PsiElement(whitespace)(' ')
          PsiElement(::)('::')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(GenericIdentifier)
                PsiElement(identifier)('a')
                PsiElement(whitespace)(' ')
            PsiElement(->)('->')
            PsiElement(whitespace)(' ')
            PSASTWrapperElement(Type)
              PSASTWrapperElement(TypeAtom)
                PSASTWrapperElement(GenericIdentifier)
                  PsiElement(identifier)('b')
                  PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(TypeInstanceDeclaration)
        PsiElement(instance)('instance')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(GenericIdentifier)
          PsiElement(identifier)('coerceRefl')
          PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(Qualified)
            PSProperNameImpl(ProperName)
              PsiElement(proper name)('Coerce')
              PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(ValueDeclaration)
          PsiElement(identifier)('coerce')
          PsiElement(whitespace)(' ')
          PsiElement(identifier)('a')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(NamedBinder)
            <empty list>
          PsiElement(=)('=')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(Var)
                PSASTWrapperElement(Qualified)
                  PsiElement(identifier)('a')
                  PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(TypeInstanceDeclaration)
        PsiElement(instance)('instance')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(GenericIdentifier)
          PsiElement(identifier)('coerceShow')
          PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(Qualified)
            PSProperNameImpl(ProperName)
              PsiElement(proper name)('Show')
              PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PsiElement(=>)('=>')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(ClassName)
          PSASTWrapperElement(Qualified)
            PSProperNameImpl(ProperName)
              PsiElement(proper name)('Coerce')
              PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(TypeConstructor)
            PSASTWrapperElement(Qualified)
              PSProperNameImpl(ProperName)
                PsiElement(proper name)('String')
                PsiElement(whitespace)(' ')
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(ValueDeclaration)
          PsiElement(identifier)('coerce')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(NamedBinder)
            <empty list>
          PsiElement(=)('=')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(Var)
                PSASTWrapperElement(Qualified)
                  PsiElement(identifier)('show')
                  PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('main')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('log')
                PsiElement(whitespace)(' ')
            PSASTWrapperElement(StringLiteral)
              PsiElement(string)('"Done"')