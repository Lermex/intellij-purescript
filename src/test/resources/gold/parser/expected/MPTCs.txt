Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('Main')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Prelude')
        PsiElement(whitespace)('\n')
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Control')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Monad')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Eff')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Console')
        PsiElement(whitespace)(' ')
        PsiElement(()('(')
        PSASTWrapperElement(PositionedDeclarationRef)
          PSASTWrapperElement(ValueRef)
            PsiElement(identifier)('log')
        PsiElement())(')')
        PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeClassDeclaration)
        PsiElement(class)('class')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('NullaryTypeClass')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(Implies)
          <empty list>
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(TypeDeclaration)
          PsiElement(identifier)('greeting')
          PsiElement(whitespace)(' ')
          PsiElement(::)('::')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeConstructor)
                PSASTWrapperElement(Qualified)
                  PsiElement(proper name)('String')
                  PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeInstanceDeclaration)
        PsiElement(instance)('instance')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('nullaryTypeClass')
        PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('NullaryTypeClass')
          PsiElement(whitespace)(' ')
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(ValueDeclaration)
          PsiElement(identifier)('greeting')
          PsiElement(whitespace)(' ')
          PsiElement(=)('=')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(StringLiteral)
                PsiElement(string)('"Hello, World!"')
                PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeClassDeclaration)
        PsiElement(class)('class')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('Coerce')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(TypeVar)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(TypeVar)
            PsiElement(identifier)('b')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(Implies)
          <empty list>
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(TypeDeclaration)
          PsiElement(identifier)('coerce')
          PsiElement(whitespace)(' ')
          PsiElement(::)('::')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeVar)
                PsiElement(identifier)('a')
                PsiElement(whitespace)(' ')
            PsiElement(->)('->')
            PsiElement(whitespace)(' ')
            PSASTWrapperElement(Type)
              PSASTWrapperElement(TypeAtom)
                PSASTWrapperElement(TypeVar)
                  PsiElement(identifier)('b')
                  PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeInstanceDeclaration)
        PsiElement(instance)('instance')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('coerceRefl')
        PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('Coerce')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(TypeVar)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(TypeVar)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(ValueDeclaration)
          PsiElement(identifier)('coerce')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Binder)
            PSASTWrapperElement(VarBinder)
              PsiElement(identifier)('a')
              PsiElement(whitespace)(' ')
          PsiElement(=)('=')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(Var)
                PSASTWrapperElement(Qualified)
                  PsiElement(identifier)('a')
                  PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeInstanceDeclaration)
        PsiElement(instance)('instance')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('coerceShow')
        PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('Show')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(TypeVar)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PsiElement(=>)('=>')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(ClassName)
          PSASTWrapperElement(Qualified)
            PsiElement(proper name)('Coerce')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(TypeVar)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(TypeConstructor)
            PSASTWrapperElement(Qualified)
              PsiElement(proper name)('String')
              PsiElement(whitespace)(' ')
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(ValueDeclaration)
          PsiElement(identifier)('coerce')
          PsiElement(whitespace)(' ')
          PsiElement(=)('=')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(Var)
                PSASTWrapperElement(Qualified)
                  PsiElement(identifier)('show')
                  PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('main')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('log')
                PsiElement(whitespace)(' ')
            PSASTWrapperElement(StringLiteral)
              PsiElement(string)('"Done"')